pulsar-broker/src/main/java/com/yahoo/pulsar/broker/loadbalance/impl/LinuxBrokerHostUsageImpl.java

```diff
@@ -0,0 +1,237 @@
+        double totalNicUsageTx = getTotalNicUsageTxKb(nics);
+        double totalNicUsageRx = getTotalNicUsageRxKb(nics);
+        double totalCpuLimit = getTotalCpuLimit();
+        CpuStat cpuStat = getTotalCpuUsage();
+        SystemResourceUsage usage = new SystemResourceUsage();
+        long now = System.currentTimeMillis();
+
+        if (lastCollection == 0L) {
+            usage.setMemory(getMemUsage());
+            usage.setBandwidthIn(new ResourceUsage(0d, totalNicLimit));
+            usage.setBandwidthOut(new ResourceUsage(0d, totalNicLimit));
+            usage.setCpu(new ResourceUsage(0d, totalCpuLimit));
+        } else {
+            double elapsedSeconds = (now - lastCollection) / 1000d;
+            double nicUsageTx = (totalNicUsageTx - lastTotalNicUsageTx) / elapsedSeconds;
+            double nicUsageRx = (totalNicUsageRx - lastTotalNicUsageRx) / elapsedSeconds;
+
+            Long cpuTimeDiff = cpuStat.getTotalTime() - lastCpuStat.getTotalTime();
+            Long cpuUsageDiff = cpuStat.getUsage() - lastCpuStat.getUsage();
+            double cpuUsage = ((double) cpuUsageDiff / (double) cpuTimeDiff) * totalCpuLimit;
+
+            usage.setMemory(getMemUsage());
+            usage.setBandwidthIn(new ResourceUsage(nicUsageRx, totalNicLimit));
+            usage.setBandwidthOut(new ResourceUsage(nicUsageTx, totalNicLimit));
+            usage.setCpu(new ResourceUsage(cpuUsage, totalCpuLimit));
+        }
+
+        lastTotalNicUsageTx = totalNicUsageTx;
+        lastTotalNicUsageRx = totalNicUsageRx;
+        lastCpuStat = cpuStat;
+        lastCollection = System.currentTimeMillis();
+        this.usage = usage;
+    }
+
+    private double getTotalCpuLimit() {
+        return (double) (100 * Runtime.getRuntime().availableProcessors());
+    }
+
+    /**
+     * Reads first line of /proc/stat to get total cpu usage.
+     * <pre>
+     *     cpu  user   nice system idle    iowait irq softirq steal guest guest_nice
+     *     cpu  317808 128  58637  2503692 7634   0   13472   0     0     0
+     * </pre>
+     * Line is split in "words", filtering the first.
+     * The sum of all numbers give the amount of cpu cycles used this far.
+     * Real CPU usage should equal the sum substracting the idle cycles,
+     * this would include iowait, irq and steal.
+     */
+    private CpuStat getTotalCpuUsage() {
+        try {
+            String[] words = Files.lines(Paths.get("/proc/stat"))
+                    .findFirst()
+                    .get().split("\\s+");
+
+            long total = Arrays.stream(words)
+                    .filter(s -> !s.contains("cpu"))
+                    .mapToLong(Long::parseLong)
+                    .sum();
+
+            long idle = Long.parseLong(words[4]);
+
+            return new CpuStat(total, total - idle);
+        } catch (IOException e) {
+            LOG.error("Failed to read CPU usage from /proc/stat", e);
+            return new CpuStat(0L, 0L);
```

Comment:
I think we'd get "divide by 0 error" at line 88, since the CPU time diff would be 0